openapi/src/main/java/io/micronaut/openapi/visitor/OpenApiExtraSchemaVisitor.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.openapi.visitor;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.order.Ordered;
+import io.micronaut.inject.ast.ClassElement;
+import io.micronaut.inject.visitor.TypeElementVisitor;
+import io.micronaut.inject.visitor.VisitorContext;
+import io.micronaut.openapi.annotation.OpenAPIExtraSchema;
+import io.swagger.v3.oas.models.media.Schema;
+
+import javax.annotation.processing.SupportedOptions;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static io.micronaut.core.util.ArrayUtils.isEmpty;
+import static io.micronaut.core.util.ArrayUtils.isNotEmpty;
+import static io.micronaut.openapi.visitor.ConfigUtils.isExtraSchemasEnabled;
+import static io.micronaut.openapi.visitor.ConfigUtils.isOpenApiEnabled;
+import static io.micronaut.openapi.visitor.ConfigUtils.isSpecGenerationEnabled;
+import static io.micronaut.openapi.visitor.ElementUtils.stringValue;
+import static io.micronaut.openapi.visitor.OpenApiConfigProperty.MICRONAUT_OPENAPI_ENABLED;
+import static io.micronaut.openapi.visitor.OpenApiModelProp.PROP_NAME;
+import static io.micronaut.openapi.visitor.SchemaDefinitionUtils.computeDefaultSchemaName;
+import static io.micronaut.openapi.visitor.SchemaDefinitionUtils.getSchemaDefinition;
+import static io.micronaut.openapi.visitor.SchemaUtils.COMPONENTS_SCHEMAS_PREFIX;
+import static io.micronaut.openapi.visitor.SchemaUtils.resolveSchemas;
+import static io.micronaut.openapi.visitor.Utils.resolveOpenApi;
+
+/**
+ * A {@link TypeElementVisitor} that builds the extra Open API schema definitions included by @{@link OpenAPIExtraSchema}
+ * at the compile time.
+ *
+ * @since 6.12.0
+ */
+@Internal
+@SupportedOptions(MICRONAUT_OPENAPI_ENABLED)
+public class OpenApiExtraSchemaVisitor implements TypeElementVisitor<Object, Object> {
```

Comment:
override `getSupportedAnnotationNames` and provide the annotation names this processes

Reply:
the first generic argument should also be `OpenApiExtraSchema`

Reply:
probably because it needs to be the repeatable wrapper `TypeElementVisitor<OpenAPIExtraSchemas, Object>`

Reply:
it's true. Fixed